pinot-core/src/main/java/org/apache/pinot/core/segment/memory/PinotByteBuffer.java

```diff
@@ -327,8 +327,11 @@ public void flush() {
 
   @Override
   protected void release() {
-    if (((DirectBuffer) _buffer).cleaner() != null) {
-      ((DirectBuffer) _buffer).cleaner().clean();
+    if (CleanerUtil.UNMAP_SUPPORTED) {
```

Comment:
If UNMAP is not supported, are we going to leak resources?

Reply:
Yes, shall we fatal the program?

Reply:
Why would we get to this code block?

Reply:
It’s possible that users may not have permission, e.g. security manager not grant reflectPermission

Reply:
I would recommend throw exception to abort the program because this will cause severe resource leak (unable to offload any segment)

Reply:
does this occur in heap and offHeap load mode?

Reply:
@kishoreg It occurs in both modes

https://stackoverflow.com/questions/36077641/java-when-does-direct-buffer-released
After some research, seems like the direct memory will be freed eventually, but there is no guarantee on how long it takes, and the JVM might run out of direct memory.
I think it is okay to log an error and keep the program running, but let's make sure in normal case it can free the direct memory properly.

Reply:
In normal case, this code will just incur different methods to clean DirectBuffer in different java runtime.